### ISSUE TITLE
Download through artifact caching proxy if it is enabled

### ISSUE THREAD

**Issue by MarkEWaite** [jenkins-infra/backend-extension-indexer#110]
### What feature do you want to see added?

The [Jenkins artifact repository](https://repo.jenkins-ci.org) is cached by the artifact caching proxy on ci.jenkins.io and its friends.  That allows us to copy files from a local cache rather than repeatedly requesting the same files over the public internet.

Unfortunately, the downloads from this source code are not using the artifact caching proxy.  Since this source code downloads the source files of all plugins and the binary files of all plugins, we are wasting bandwidth to the [Jenkins artifact repository](https://repo.jenkins-ci.org) 

### Upstream changes

Check with the Pipeline steps doc generator that the added caching does not break the generator.

**Comment by dduportal** [comment]
> @dduportal is it safe to switch from ARTIFACT_CACHING_PROXY_ORIGIN to ARTIFACT_CACHING_PROXY_SERVERID?

Absolutely. I have no idea where does `ARTIFACT_CACHING_PROXY_ORIGIN` comes from. Looks like something in https://github.com/jenkins-infra/javadoc/blob/eefddd259106d7048f7a2d843038e2d62a0ec0e4/Jenkinsfile#L35 but it is clearly a wrong thing.

`ARTIFACT_CACHING_PROXY_SERVERID` is coincidentally the URL, but even if it is working, it should not (Maven does not like using special characters in the `id` ). I would not use it unless temporary setup.

What I read, is that the `withArtifactCachingProxy()` shared library could add a the current URL in the env, along with the path to maven settings files (e.g. adding an element in https://github.com/jenkins-infra/pipeline-library/blob/e6b87f357b25159eb6161d1d168656312119d2f0/vars/infra.groovy#L204 and that should be enough for you to consume here). WDYT?
